table/Table.jsx

```diff
@@ -90,6 +90,8 @@ function handleCellClick(e: SyntheticEvent<HTMLDivElement>) {
   e.stopPropagation();
 }
 
+const TableScrollContext = React.createContext();
```

Comment:
Nit: we can avoid triggering an empty render (`if (!scrollContext) return false`) by providing a default value to TableScrollContext.

```suggestion
const TableScrollContext = React.createContext({
  addPinRef: () => null,
  pinElements: {
    top: [],
    left: [],
    right: [],
  },
  scrollLeft: 0,
  rightOffset: 0,
  scrollTop: 0,
  isPositionStickySupported: false,
});
```

Reply:
> NIT: Let's keep the table pinning logic as consolidated and separated from other table logic as possible. We can group up scrollContextData related definitions and callbacks into a factory. This would keep the scroll polyfill functionality isolated from the rest of the table implementation.
> 
> ```ts
> function createScrollData() {
>   const scrollContextData = { ... };
>   function onScroll ...
>   function updatePinPositions ...
>   function addPinRef ...
>   const outerElementType
> } => ({ scrollContextData, outerElementType, ... })
> ```
> 
> Consider also breaking out the TableScrollContext and it's related content out into a separate file.

This is a lot cleaner, thank you!